Improve documentation details and score (#187)
dart-lang/shelf: README.md

* Trivially mix and match synchronous and asynchronous processing.

> * Flexibility to return a simple string or a byte stream with the same model.

Reviewer: The last bullet isn't parallel with the preceding ones. :) Maybe:

```suggestion
* Using the same model, return either a simple string or a byte stream.
```

> See [Write HTTP servers](https://dart.dev/tutorials/server/httpserver) for more

Reviewer: We might change the page name/location, so maybe use a more generic name:

```suggestion
See the [Dart HTTP server documentation](https://dart.dev/tutorials/server/httpserver) for more
```
